sdks/nodejs/src/agonesSDK.js

```diff
@@ -23,7 +23,20 @@ class AgonesSDK {
 		this.healthStream = undefined;
 		this.emitters = [];
 	}
-	async close(){
+
+	async connect() {
+		return new Promise((resolve, reject) => {
+			this.client.waitForReady((new Date).getTime() + 30000, (error) => {
```

Comment:
I would be tempted to add the timeout to the constructor with a default value. E.g.
```
constructor(timeout = 30000) {
    this.timeout = timeout
...
use later as Date.now() + this.timeout
```
Or if you like pass in as seconds and multiply by 1000 in the constructor

Reply:
All other SDKs timeout by 30 seconds - we control the SDK server, so not personally seeing a good reason to make this configurable. I can't think of a use case in which someone would want to make this larger/smaller?